src/ds3_connection.c

```diff
@@ -95,6 +96,7 @@ void ds3_connection_release(ds3_connection_pool* pool, ds3_connection* connectio
     g_mutex_lock(&pool->mutex);
     curl_easy_reset(connection);
     pool->tail = _pool_inc(pool->tail, pool->num_connections);
+    printf("  release connection [%d]\n", pool->tail);
```

Comment:
ds3_log_message requires the client, which we don't have here, and this msg will come back out before final commit.  I just added this to visually verify that the lock / unlocking is happening asynchronously